### PR TITLE
DM-430: Modify measurement framework to log plugins error in individual logs

### PR DESCRIPTION
**Pull request by rearmstr**

### CODE REVIEW

File: include/lsst/meas/base/PsfFlux.h [hunk 1]
```diff
@@ -78,7 +78,8 @@ class PsfFluxAlgorithm : public SimpleAlgorithm {
     /// The control object contains the configuration parameters for this algorithm.
     typedef PsfFluxControl Control;
 
-    PsfFluxAlgorithm(Control const & ctrl, std::string const & name, afw::table::Schema & schema);
+    PsfFluxAlgorithm(Control const & ctrl, std::string const & name, afw::table::Schema & schema,
+                           std::string const & logName = "");
```
Comment (rearmstr) [inline include/lsst/meas/base/PsfFlux.h]: indentation should match the beginning of the parentheses.

File: python/lsst/meas/base/algorithm.cc [hunk 2]
```diff
@@ -41,16 +41,19 @@ PYBIND11_PLUGIN(algorithm) {
...
     py::class_<SimpleAlgorithm, std::shared_ptr<SimpleAlgorithm>, SingleFrameAlgorithm> clsSimpleAlgorithm(
-            mod, "SimpleAlgorithm");
+            mod, "SimpleAlgorithm", py::multiple_inheritance());
 
     /* Members */
     python::declareAlgorithm<SingleFrameAlgorithm>(clsSingleFrameAlgorithm);
 
     clsSimpleAlgorithm.def("measureForced", &SimpleAlgorithm::measureForced,
         "measRecord"_a, "exposure"_a, "refRecord"_a, "refWcs"_a);
+    clsSimpleAlgorithm.def("getLogName", &SimpleAlgorithm::getLogName);
```
Comment (rearmstr) [inline python/lsst/meas/base/algorithm.cc]: I haven't learned how to wrap in pybind11 yet, but shouldn't this be a method of BaseAlgorithm?

File: python/lsst/meas/base/wrappers.py [hunk 3]
```diff
@@ -119,8 +125,7 @@ def wrapAlgorithm(Base, AlgClass, factory, executionOrder, name=None, Control=No
                                If apCorrList is non-empty then shouldApCorr is ignored.
                                If non-empty and doRegister is True then the names are added to the set
                                retrieved by getApCorrNameSet
-
-
+    @param[in] hasLogName      Plugin supports a logName as a constructor argument
```
Comment (rearmstr) [inline python/lsst/meas/base/wrappers.py]: Removed extra blank line.  Not sure what the standard is, but should be in a separate commit.

File: python/lsst/meas/base/wrappers.py [hunk 4]
```diff
@@ -401,9 +428,10 @@ def wrapTransform(transformClass):
     """
     oldInit = transformClass.__init__
 
-    def _init(self, ctrl, name, mapper):
+    def _init(self, ctrl, name, mapper, logName=None):
         if hasattr(ctrl, "makeControl"):
             ctrl = ctrl.makeControl()
+        #  logNmae signature needs to be on this Class __init__, but is not needed by the C++ plugin
```
Comment (rearmstr) [inline python/lsst/meas/base/wrappers.py]: logNmae -> logName

File: tests/testPluginLogs.py [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+        props += "log4j.appender.FA.Append=false\n"
+    props += "log4j.appender.FA.layout=PatternLayout\n"
+    props += "log4j.appender.FA.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss.SSS} %p %c %m %X%n\n"
+    props += "log4j.logger.main.a=DEBUG\n"
+    log.configure_prop(props)
+
+class RegisteredPluginsTestCase(AlgorithmTestCase, lsst.utils.tests.TestCase):
+    """
+    Test all the registered Plugins to see if their logName is set as expected.
+    Those which have the hasLogName=True  attribute will have a LogName parameter
```
Comment (rearmstr) [inline tests/testPluginLogs.py]: extra space after "True"

**Comment by pgee2000** [comment]
I responded by fixing all the comments on the pull request as per Bob Armstrong's comment.
